### ISSUE TITLE
Remove overlap of comments with private data wire

### ISSUE THREAD

**Issue by francois-normandin** [JKISoftware/JKI-State-Machine#15]
This is just an annoyance really, but the overlap between the private data wire and the comments at the top of the cases creates kinks in wires when using common quick drop shortcuts.

![image](https://user-images.githubusercontent.com/11728548/58371167-0ba8e100-7edd-11e9-9cc1-06d20a7eb260.png)

For example, if I were to bundle some data, and later decide to remove it using Ctrl-R shortcut, it will rewire with a kink for the wire.

Step to reproduce:
1- Bundle data.
![image](https://user-images.githubusercontent.com/11728548/58371660-3432da00-7ee1-11e9-9b76-b205cf754be2.png)

2- Select and do Ctrl-Space, Ctrl-R
![image](https://user-images.githubusercontent.com/11728548/58371674-53316c00-7ee1-11e9-93ad-431fcc5e6e7f.png)

**Comment by jimkring** [comment]
@francois-normandin, good point. I'm curious what ideas you might have about how we could fix this?

For example, one could (but maybe should not):

- Use a smaller font for comments (yet, this makes it harder to read)
- Limit comments to a single line (I don't see too much downside to this, although two lines is nice)
- Move comments to upper left corner, instead of in the middle (since the drop-down case selector above the comments takes up a lot of vertical space)
- Move shift registers down a little bit (yet this takes up space often used by the code)
- Increase space (ctrl+drag) between top of case and the data cluster (yet, this violates [the best practice of not making the JKI SM bigger](https://blog.jki.net/products/state-machine/3-best-practice-for-using-the-jki-state-machine-keep-the-original-size-i-e-dont-grow-the-structures))

Thanks!

**Comment by francois-normandin** [comment]
@jimkring I'm sure you already know which one I prefer :-)

Personally, I always add a few tens of pixels to my JKI SMs at the top to show the dynamic event registration terminals on the Event Structure. I get the added benefit of giving clearance to the text boxes, which is a useful place for a description of the code executing in that frame.

https://www.screencast.com/users/normandinf/folders/Snagit/media/25b49e6c-d68b-412a-9a13-e186ba53dfa0

**Comment by jimkring** [comment]
Hi @francois-normandin. OK, I think I see what you mean. Is this what you're talking about?

![image](https://user-images.githubusercontent.com/381432/58575905-f2af7100-81f7-11e9-8ed4-1e8d6819796c.png)

**Comment by francois-normandin** [comment]
👍

**Comment by jlj-ee** [comment]
Like @francois-normandin, I always expand the space at the top the event structure in any new JKI SM that I put down.
This has a couple aesthetic benefits for me:
1. Room for dynamic event terminals and no wire kinds while using quick drop actions, as noted above.
2. Event case comments can also reside above the cluster wire, as they do in non-event states. That way, they don't get in the way of any code I may have in the event case. Often, there is minimal code because I'll be calling another state, but for certain event types that must be handled in that case (like filter events) it helps a lot to not have the comments in the way. Plus, it adds some visual consistency with the non-event states.

As already noted, it's a very minor annoyance, but @francois-normandin is not alone :)

**Comment by jimkring** [comment]
Thanks for the weighing in on this @jlj-ee. I'm convinced :-)

**Comment by Tom-McQuillan** [comment]
Closed as resolved.
